core/src/main/java/jenkins/views/Header.java

```diff
@@ -0,0 +1,60 @@
+package jenkins.views;
+
+import java.util.Optional;
+
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+
+/**
+ * Extension point that provides capabilities to render a specific header.
+ *  
+ * Extend {@link PartialHeader} or {@link FullHeader} depending on the use case.
+ * 
+ * The default Jenkins header is provided as an implementation of a {@link FullHeader} 
+ * named {@link JenkinsHeader}.
+ * 
+ * All headers will provide a prioritization technique, via the usual ordinal field of 
+ * the {@link Extension} annotation.
```

Comment:
something like this maybe.
```suggestion
 * The first header located will be used, set the ordinal field on {@link Extension} to have a higher priority.
```